.github/workflows/wheel-universal.yml

```diff
@@ -2,166 +2,85 @@ name: Build universal wheel
-
     - name: Build vaex-contrib
-      if: startsWith(github.ref, 'refs/tags/contrib')
       run: |
         (cd packages/vaex-contrib && python setup.py sdist bdist_wheel); cp packages/vaex-contrib/dist/* dist
 
-    - name: Publish a Python distribution to PyPI
-      if: startsWith(github.ref, 'refs/tags/contrib')
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.pypi_password_vaex_contrib }}
-      run: |
-        twine upload dist/vaex-contrib* dist/vaex_contrib*
-
-
-
     - name: Build vaex-viz
-      if: startsWith(github.ref, 'refs/tags/viz')
       run: |
         (cd packages/vaex-viz && python setup.py sdist bdist_wheel); cp packages/vaex-viz/dist/* dist
 
-    - name: Publish a Python distribution to PyPI
-      if: startsWith(github.ref, 'refs/tags/viz')
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.pypi_password_vaex_viz }}
-      run: |
-        twine upload dist/vaex-viz* dist/vaex_viz*
-
     - name: Build vaex-astro
-      if: startsWith(github.ref, 'refs/tags/astro')
       run: |
         (cd packages/vaex-astro && python setup.py sdist bdist_wheel); cp packages/vaex-astro/dist/* dist
 
-    - name: Publish a Python distribution to PyPI
-      if: startsWith(github.ref, 'refs/tags/astro')
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.pypi_password_vaex_astro }}
-      run: |
-        twine upload dist/vaex-astro* dist/vaex_astro*
-
     - name: Build vaex-hdf5
-      if: startsWith(github.ref, 'refs/tags/hdf5')
       run: |
         (cd packages/vaex-hdf5 && python setup.py sdist bdist_wheel); cp packages/vaex-hdf5/dist/* dist
 
-    - name: Publish a Python distribution to PyPI
-      if: startsWith(github.ref, 'refs/tags/hdf5')
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.pypi_password_vaex_hdf5 }}
-      run: |
-        twine upload dist/vaex-hdf5* dist/vaex_hdf5*
-
     - name: Build vaex-server
-      if: startsWith(github.ref, 'refs/tags/server')
       run: |
         (cd packages/vaex-server && python setup.py sdist bdist_wheel); cp packages/vaex-server/dist/* dist
 
-    - name: Publish a Python distribution to PyPI
-      if: startsWith(github.ref, 'refs/tags/server')
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.pypi_password_vaex_server }}
-      run: |
-        openssl sha256 dist/*
-        twine upload dist/vaex-server* dist/vaex_server*
-
-    - uses: actions/upload-artifact@v1
+    - uses: actions/upload-artifact@v4
       with:
-        name: distributions
+        name: distributions-universal
         path: ./dist
+
+    # https://github.com/pypa/gh-action-pypi-publish#trusted-publishing
+    - name: Publish package distributions to PyPI
+      uses: pypa/gh-action-pypi-publish@v1.9.0
+      if: startsWith(github.ref, 'refs/tags')
+      with:
+        skip-existing: true
```

Comment:
all you need to do for this to work is add github.com/vaexio/vaex as trusted publisher https://pypi.org/manage/project/vaex/settings/publishing (and to the other vaex pypi projects)

Reply:
This now fails, because it will try to release all packages (even though I only pushed the vaex-core tag).

I guess if we only do the `cp` commands when the specific tag is used, we can make this work. What do you think?